Inferences/GeneralInduction.cpp

```diff
@@ -346,25 +346,27 @@ void GeneralInduction::generateClauses(
 
   // Resolve all induction clauses with the main and side literals
   auto resSubst = ResultSubstitution::fromSubstitution(&subst, 0, 1);
+  mainQuery.substitution = resSubst;
+  // Be aware that we change mainLit and sideLitQrPairs here irreversibly
```

Comment:
This comment reminded me - are there any pointer/reference arguments in any of the methods you introduce that are not supposed to be changed? If yes, it would be good to mark them const.